pkg/controller/kubefedcluster/controller.go

```diff
@@ -257,6 +260,9 @@ func (cc *ClusterController) updateIndividualClusterStatus(cluster *fedv1b1.Kube
 	if err := cc.client.UpdateStatus(context.TODO(), cluster); err != nil {
 		klog.Warningf("Failed to update the status of cluster %q: %v", cluster.Name, err)
 	}
+
+	metrics.ClusterHealthStatusDurationFromStart(clusterHealthStatusStart)
```

Comment:
Why didn't you use a defer like you did [here](https://github.com/kubernetes-sigs/kubefed/pull/1196/files#diff-0dfac9e46f796a7e5d47adce78da8e0eR133)?

Reply:
It wasn't intentional, I started using them in a way. I now adapted to run as a `defer` call